src/service/data_products/filtering.py

```diff
@@ -3,26 +3,69 @@
 data products like genome attributes or samples.
 """
 
+from abc import ABC, abstractmethod
 from dateutil import parser
 from pydantic import BaseModel, Field
-from src.common.product_models.columnar_attribs_common_models import ColumnType
+from src.common.product_models.columnar_attribs_common_models import ColumnType, FilterStrategy
 
 from types import NotImplementedType
-from typing import Any, Self
+from typing import Annotated, Any, Self
 
 
 class SearchQueryPart(BaseModel):
+    variable_assignments: Annotated[dict[str, str], Field(
```

Comment:
should be Annotated[dict[str, str] | `None`, Field(...)] = None?

Reply:
Fixed